app/drivers/distributed_manager.go

```diff
@@ -567,5 +673,41 @@ func (d *DistributedManager) executeInstanceCleanup(ctx context.Context, pool *p
 		logr.WithError(err).Errorf("distributed dlite: failed to delete %s instances of pool=%q", cleanupType, pool.Name)
 	}
 
+	go d.destroyCapacity(ctx, instances)
```

Comment:
do you need to do this ? only `destroyCapacityFromReservation` should be responsible to to delete the capacity and instance (if it exists) associated to it.